gtk/src/toga_gtk/widgets/multilinetextinput.py

```diff
@@ -91,6 +91,13 @@ def set_value(self, value):
             else:
                 self.native_textview.set_buffer(self.buffer)
 
+        if GTK_VERSION < (4, 0, 0):  # pragma: no-cover-if-gtk4
+            # Wait until the text change is fully completed
+            while Gtk.events_pending():
+                Gtk.main_iteration_do(blocking=False)
+        else:  # pragma: no-cover-if-gtk3
+            pass
```

Comment:
Two notes here:

Firstly, the missing GTK branch can be populated with:
```
while GLib.main_context_default().pending():
    GLib.main_context_default().iteration(may_block=False)
```

Secondly - I don't know if we have other places where this might be needed ,but it is a sufficiently "core" piece of functionality that it would be worth wrapping as a general  `flush_gtk_events()` method on the base GTK widget class, and then invoking that method here.